include/zephyr/timing/timing.h

```diff
@@ -14,33 +14,246 @@
 extern "C" {
 #endif
 
+/**
+ * @brief Timing Measurement APIs
+ * @defgroup timing_api Timing Measurement APIs
+ * @ingroup os_services
+ */
+
+/**
+ * @brief SoC specific Timing Measurement APIs
+ * @defgroup timing_api_soc SoC specific Timing Measurement APIs
```

Comment:
would it benefit from being mounted somewhere in the Sphinx doc? Probably the "Executing Time Functions" page?

Reply:
I totally forgot about that page... added these groups to that doc.